### PR TITLE
Add waveform to voice message preview UI

### PR DESCRIPTION
**Pull request by jonnyandrew**
## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Technical
- [ ] Other :

## Content

Add waveform to voice message preview UI.

Seeking and playback progress to follow.

## Motivation and context

- Part of https://github.com/vector-im/element-meta/issues/2104

## Tested devices

- [x] Physical
- [x] Emulator
- OS version(s): 13

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-x-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR

### CODE REVIEW

**Review by jmartinesp — APPROVED** [review]
LGTM, thanks! I found a minor issue though: if you are playing the recorded message and either delete it or send it, it will continue playing.

**Comment by jonnyandrew** [comment]
Thanks for testing! The issue with the send button is known but not the delete button - I'll fix that in a follow up

File: libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/components/media/FakeWaveformFactory.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * limitations under the License.
+ */
+
+package io.element.android.libraries.designsystem.components.media
+
+import kotlinx.collections.immutable.ImmutableList
+import kotlinx.collections.immutable.toPersistentList
+import kotlin.random.Random
+
+object FakeWaveformFactory {
```
Comment (julioromano) [inline libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/components/media/FakeWaveformFactory.kt:23]: It seems to me the codebase tends to prefer top level function for these things: `fun aFakeWaveform()`

Comment (jonnyandrew) [reply libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/components/media/FakeWaveformFactory.kt:23]: Ok, I'll fix in a follow up

**Pull request merged**